core/src/com/ericc/the/game/map/Dungeon.java

```diff
@@ -43,34 +44,39 @@ public void changeLevel(int levelNumber, InitialPlayerPosition initialPlayerPosi
         }
         currentLevelNumber = levelNumber;
         loadProgress(newLevel);
-        placePlayer(initialPlayerPosition);
+        placePlayer(initialPlayerPosition, newLevel.getMap());
     }
 
     private void loadProgress(Level level) {
-        CurrentMap.setMap(level.getMap());
         for (Entity entity : level.getEntities()) {
             engines.addEntity(entity);
         }
     }
 
     private void saveProgress() {
-        Family family = Family.all(PositionComponent.class).exclude(PlayerComponent.class).get();
-        ArrayList<Entity> entities = ImmutableArrayUtils.toArrayList(engines.getEntitiesFor(family));
-        levels.put(currentLevelNumber, new Level(CurrentMap.map, entities));
+        Family notPlayersFamily = Family.all(PositionComponent.class).exclude(PlayerComponent.class).get();
+        Family playersFamily = Family.all(PlayerComponent.class).get();
+        ArrayList<Entity> notPlayers = ImmutableArrayUtils.toArrayList(engines.getEntitiesFor(notPlayersFamily));
+        ImmutableArray<Entity> players = engines.getEntitiesFor(playersFamily);
+        Map currentMap = players.get(0).getComponent(PositionComponent.class).map;
+        levels.put(currentLevelNumber, new Level(currentMap, notPlayers));
 
-        engines.removeFamily(family);
+        engines.removeFamily(notPlayersFamily);
     }
 
-    private void placeEntity(Entity entity, GridPoint2 desiredPosition) {
+    private void placeEntity(Entity entity, GridPoint2 desiredPosition, Map map) {
         PositionComponent entityPosition = Mappers.position.get(entity);
 
         for (GridPoint2 move : Moves.moves) {
             int x = desiredPosition.x + move.x;
             int y = desiredPosition.y + move.y;
 
-            if (levels.get(currentLevelNumber).getMap().isPassable(x, y)) {
+            if (map.isFloor(x, y)) {
+                entityPosition.map.entityMap.remove(new GridPoint2(entityPosition.x, entityPosition.y));
```

Comment:
Domyslilem sie co sie tu dzieje, ale skomentuj to -> (podmiana mapy, usuniecie z poprzedniej etc)